common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/NativeAuthRequestProvider.kt

```diff
@@ -157,6 +165,108 @@ class NativeAuthRequestProvider(private val config: NativeAuthOAuth2Configuratio
     }
     //endregion
 
+    //region /signup/start
+    internal fun createSignUpStartRequest(
+        commandParameters: SignUpStartCommandParameters
+    ): SignUpStartRequest {
+        LogSession.logMethodCall(TAG, "${TAG}.createSignUpStartRequest")
+
+        return SignUpStartRequest.create(
+            username = commandParameters.username,
+            attributes = commandParameters.userAttributes,
+            challengeType = config.challengeType,
+            clientId = config.clientId,
+            requestUrl = signUpStartEndpoint,
+            headers = getRequestHeaders()
+        )
+    }
+
+    internal fun createSignUpUsingPasswordStartRequest(
+        commandParameters: SignUpStartUsingPasswordCommandParameters
+    ): SignUpStartRequest {
+        LogSession.logMethodCall(TAG, "${TAG}.createSignUpUsingPasswordStartRequest")
+
+        if (commandParameters.password.isEmpty() || commandParameters.password.all { it.isWhitespace() })
```

Comment:
Do you also need to validate the username/email?

Reply:
We check that username is not empty in `SignUpStartRequest.create` method.